halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Node.java

```diff
@@ -61,6 +68,36 @@ public boolean matchesToRoot(NodeFilter filter) {
   @JsonIgnore
   public abstract NodeReference getReference();
 
+  @JsonIgnore
+  public List<String> fieldOptions(ProblemSetBuilder problemSetBuilder, String fieldName) {
+    if (fieldName == null || fieldName.isEmpty()) {
+      throw new IllegalArgumentException("Input fieldName may not be empty");
+    }
+
+    log.info("Looking for options for field " + fieldName + " in node " + getNodeName() + " for type " + getClass());
+    String fieldOptions = fieldName + "Options";
+    Method optionsMethod = null;
+
+    try {
+      optionsMethod = this.getClass().getDeclaredMethod(fieldOptions, ProblemSetBuilder.class);
+      optionsMethod.setAccessible(true);
+
+      return (List<String>) optionsMethod.invoke(this, problemSetBuilder);
+    } catch (IllegalAccessException | InvocationTargetException e) {
+      log.warn("Failed to call " + fieldOptions + "() on " + this.getClass());
+      e.getCause().printStackTrace();
+
+      throw new RuntimeException(e.getCause());
```

Comment:
Won't printing the cause and then re-throwing a `RuntimeException` with that cause print the stack trace twice?